roadmap/implementers-guide/src/node/backing/prospective-parachains.md

```diff
@@ -0,0 +1,111 @@
+
+Prospective Parachains is primarily an implementation of fragment trees. See the
+"Fragment Trees" section. (TODO) (Also see the Inclusion Emulator page.)
+
+It also handles concerns such as:
+
+  - the relay-chain being forkful
+  - session changes
+  - predicting validator group assignments
+
+## Messages
+
+### Incoming
+
+- `ActiveLeaves`
+  - Notification of a change in the set of active leaves.
+- `ProspectiveParachainsMessage::IntroduceCandidate`
+  - Informs the subsystem of a new candidate.
+  - Sent by the Backing Subsystem when it is importing a statement for a
+    new candidate.
+- `ProspectiveParachainsMessage::CandidateSeconded`
+  - Informs the subsystem that a previously introduced candidate has
+    been seconded.
+  - Sent by the Backing Subsystem when it is importing a statement for a
+    new candidate after it sends `IntroduceCandidate`, if that wasn't
+    rejected by Prospective Parachains.
+- `ProspectiveParachainsMessage::CandidateBacked`
+  - Informs the subsystem that a previously introduced candidate has
+    been backed.
+  - Sent by the Backing Subsystem after it successfully imports a
+    statement for the first time.
+- `ProspectiveParachainsMessage::GetBackableCandidate`
+  - Get a backable candidate hash for a given parachain, under a given
+    relay-parent hash, which is a descendant of given candidate hashes.
+  - Sent by the Provisioner when requesting backable candidates, when
+    selecting candidates for a given relay-parent.
+- `ProspectiveParachainsMessage::GetHypotheticalFrontier`
+  - Gets the hypothetical frontier membership of candidates with the
+    given properties under the specified active leaves' fragment trees.
+  - Sent by the Backing Subsystem when checking whether a candidate can
+    be seconded based on its hypothetical frontiers.
+- `ProspectiveParachainsMessage::GetTreeMembership`
+  - Gets the membership of the candidate in all fragment trees.
+  - Sent by the Backing Subsystem when it updates the candidates
+    seconded at various depths under active leaves.
+- `ProspectiveParachainsMessage::GetMinimumRelayParents`
+  - Gets the minimum accepted relay-parent number for each para in the
+    fragment tree for the given relay-chain block hash.
+  - That is, this returns the minimum relay-parent block number in the
+    same branch of the relay-chain which is accepted in the fragment
+    tree for each para-id.
+  - Sent by the Backing, Statement Distribution, and Collator Protocol
+    subsystems when activating leaves in the implicit view.
+- `ProspectiveParachainsMessage::GetProspectiveValidationData`
+  - Gets the validation data of some prospective candidate. The
+    candidate doesn't need to be part of any fragment tree.
+  - Sent by the Collator Protocol subsystem (validator side) when
+    handling a fetched collation result.
+
+### Outgoing
+
+- `RuntimeApiRequest::StagingParaBackingState`
+  - Gets the backing state of the given para.
```

Comment:
Good catch. I'm wondering if this should be removed, as the doc says `This is a staging API that will not be available on production runtimes`. @rphmeier can you please explain a bit what the purpose of this is and why it will be removed?

Reply:
It won't be removed. This means that polkadot and kusama runtimes will not implement it initially (thus having async backing disabled).

This is the key runtime call, when we stabilize async backing, `staging` part will be removed and the method will transition to stable `v5` (or smth)

My comment means we should add an explanation that backing state is a pair of constraints and current core state (if there is a candidate occupying the code whose availability isn't reached yet)

Reply:
https://github.com/paritytech/polkadot/blob/1e06eb8ac76fcaf523a08ac7f6a10c310571fe0f/runtime/parachains/src/runtime_api_impl/vstaging.rs#L30

Reply:
Thank you, that makes a lot more sense. I just got confused by the comment.